src/cluster.c

```diff
@@ -596,6 +601,9 @@ void clusterInit(void) {
     server.cluster->state = CLUSTER_FAIL;
     server.cluster->size = 1;
     server.cluster->todo_before_sleep = 0;
+    server.cluster->moved_slot_since_sleep = MOVED_SLOT_NONE;
+    server.cluster->moved_slot_channel =
+        createObject(OBJ_STRING, sdsnew("__redis__:moved"));
```

Comment:
We have these system channels already:

* `__redis__:invalidate` is used in client-side caching.
* `__keyspace@0__:*` and `__keyevent@0__:*` used in keyspace notifications. These are dynamic and contain the db number (0) in the prefix. After the prefix is a key or a command.

I don't mind restricting channels with the `__redis__:` prefix but I'm not think it's required either. It can be done later. It's a breaking change but probably an acceptable one.

Reply:
Agreed it is an orthogonal change that can be done separately. Timing wise though, it would be nice if we could introduce this breaking change (tightening up system channels) as part 7.0 release. I will file a bug to track the discussion independently.